server/src/main/java/io/crate/execution/ddl/tables/MappingUtil.java

```diff
@@ -134,41 +134,49 @@ public static Map<String, Object> createMapping(AllocPosition allocPosition,
         return mapping;
     }
 
-    public static Map<String, Object> createMappingForDroppedCols(DocTableInfo tableInfo,
-                                                                  HashMap<ColumnIdent, List<Reference>> tree) {
-        Map<String, Map<String, Object>> propertiesMap =
-            createPropertiesMap(AllocPosition.forTable(tableInfo), null, tree);
-
-        Map<String, Object> mapping = new HashMap<>();
-        mapping.put("_meta", Map.of());
-        mapping.put("properties", propertiesMap);
-        return mapping;
-    }
-
-
     /**
-     * Creates a mapping containing only column names, can be nested.
+     * Similar to {@link #toProperties(AllocPosition, HashMap)} but instead of adding the full schema for a column, this only adds the names.
+     *
+     * Example output:
+     * {
+     *     col1: {
+     *        * optional, only for nested objects *
+     *        properties: {
+     *            nested_col1: {...},
+     *            nested_col2: {...},
+     *        }
+     *     },
+     *     col2: {...}
+     * }
      */
-    public static Map<String, Object> createMappingToRemove(HashMap<ColumnIdent, List<Reference>> tree,
-                                                            @Nullable ColumnIdent currentNode) {
-        List<Reference> children = tree.get(currentNode);
+    public static Map<String, Object> toNamesOnlyProperties(HashMap<ColumnIdent, List<Reference>> tree) {
+        return toNamesOnlyProperties(tree, null);
+    }
```

Comment:
Sure we could merge the two, but imho the public API shouldn't have that extra parameter because it doesn't make sense to provide an arbitrary column/node.